src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java

```diff
@@ -267,6 +270,22 @@ else if (paymentGroup.getDisbursementType().getCode().equals(PdpConstants.Disbur
         }
     }
     
+    private String getCheckYear(PaymentGroup paymentGroup) {
+        String checkYear = KFSConstants.getDashSubAccountNumber();
+        try {
+            CheckReconciliation checkReconciliation = checkReconciliationDao.findByCheckNumber(paymentGroup.getDisbursementNbr().toString());
+            Calendar calendar = Calendar.getInstance();
+            if (ObjectUtils.isNotNull(checkReconciliation) && ObjectUtils.isNotNull(checkReconciliation.getCheckDate())) {
```

Comment:
Should the second part of this if statement be StringUtils.isNotEmpty?  Should there be an else to note an error condition?

Reply:
I added an else condition to log an error without throwing an exception. Should the log be a warning message instead?

Reply:
I added an else condition here. The object checkReconciliation.getDate() is of type java.util.date.